### ISSUE TITLE
HA - rke - tls: bad certificate

### ISSUE THREAD

**Issue by branislav-brujic** [rancher/rke#1446]
**RKE version:** v.0.2.4

**Docker version: (`docker version`,`docker info` preferred)**  18.09.2

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
ubuntu 18 LTS 4.15.0-47-generic

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
Bare metal

**cluster.yml file:**
```
nodes:
    - address: **.**.**.21
      ssh_key_path: key.pem
      user: someuser
      role:
        - controlplane
        - etcd
    - address: **.**.**.20
      ssh_key_path: key.pem
      user: someuser
      role:
        - controlplane
        - etcd
    - address: **.**.**.106
      ssh_key_path: key.pem
      user: someuser
      role:
        - worker
    - address: **.**.**.58
      ssh_key_path: key.pem
      user: someuser
      role:
        - worker

services:
  etcd:
    snapshot: true
    creation: 6h
    retention: 24h
```

**Steps to Reproduce:**
`rke --debug up --update-only --config cluster.yml`

**Results:**
```
INFO[0123] [etcd] Successfully started [rke-log-linker] container on host [ **.**.**.20] 
DEBU[0123] [remove/rke-log-linker] Checking if container is running on host [ **.**.**..20] 
DEBU[0126] [remove/rke-log-linker] Removing container on host [ **.**.**.20] 
INFO[0126] [remove/rke-log-linker] Successfully removed container on host [ **.**.**..20] 
DEBU[0126] [etcd] Successfully created log link for Container [etcd] on host [ **.**.**..20] 
INFO[0126] [etcd] Successfully started etcd plane.. Checking etcd cluster health 
DEBU[0126] [etcd] Check etcd cluster health             
DEBU[0128] Failed to get /health for host [ **.**.**..21]: Get https:// **.**.**..21:2379/health: remote error: tls: bad certificate 
DEBU[0138] Failed to get /health for host [ **.**.**..21]: Get https:// **.**.**..21:2379/health: remote error: tls: bad certificate 
DEBU[0148] Failed to get /health for host [ **.**.**..21]: Get https:// **.**.**.:2379/health: remote error: tls: bad certificate 
DEBU[0153] [etcd] Check etcd cluster health             
DEBU[0155] Failed to get /health for host [ **.**.**..20]: Get https://.20:2379/health: remote error: tls: bad certificate 
DEBU[0168] Failed to get /health for host [ **.**.**.20]: Get https:// **.**.**..20:2379/health: remote error: tls: bad certificate 
DEBU[0177] Failed to get /health for host [ **.**.**.20]: Get https:// **.**.**.20:2379/health: remote error: tls: bad certificate 
FATA[0182] [etcd] Failed to bring up Etcd Plane: [etcd] Etcd Cluster is not healthy 
```

---------------------------------------------------------
Everything works good I just cant add one new node to cluster because of this `bad certificate` issue.
I using rancher self signed certificate

**Comment by panho66** [comment]
I have similar problem

**step(s)**
- Create cluster.yml
- rke up 
- **no problem**

**Remove kube_config_cluster.yml and rke up**
- **no problem**

**Remove kube_config_cluster.yml and cluster.rkestate then rke up** 

- **error: tls: bad certificate**

DEBU[0033] [remove/rke-log-linker] Removing container on host [10.100.138.13]
INFO[0034] [remove/rke-log-linker] Successfully removed container on host [10.100.138.13]
DEBU[0034] [etcd] Successfully created log link for Container [etcd] on host [10.100.138.13]
INFO[0034] [etcd] Successfully started etcd plane.. Checking etcd cluster health
DEBU[0034] [etcd] Check etcd cluster health
DEBU[0034] Failed to get /health for host [10.100.138.45]: Get https://10.100.138.45:2379/health: remote error: tls: bad certificate
DEBU[0039] Failed to get /health for host [10.100.138.45]: Get https://10.100.138.45:2379/health: remote error: tls: bad certificate
DEBU[0045] Failed to get /health for host [10.100.138.45]: Get https://10.100.138.45:2379/health: remote error: tls: bad certificate
DEBU[0050] [etcd] Check etcd cluster health
DEBU[0050] Failed to get /health for host [10.100.138.250]: Get https://10.100.138.250:2379/health: remote error: tls: bad certificate
DEBU[0055] Failed to get /health for host [10.100.138.250]: Get https://10.100.138.250:2379/health: remote error: tls: bad certificate
DEBU[0060] Failed to get /health for host [10.100.138.250]: Get https://10.100.138.250:2379/health: remote error: tls: bad certificate
DEBU[0065] [etcd] Check etcd cluster health
DEBU[0066] Failed to get /health for host [10.100.138.13]: Get https://10.100.138.13:2379/health: remote error: tls: bad certificate
DEBU[0071] Failed to get /health for host [10.100.138.13]: Get https://10.100.138.13:2379/health: remote error: tls: bad certificate
DEBU[0076] Failed to get /health for host [10.100.138.13]: Get https://10.100.138.13:2379/health: remote error: tls: bad certificate
FATA[0081] [etcd] Failed to bring up Etcd Plane: [etcd] Etcd Cluster is not healthy

**Remove cluster.rkestate (keep original kube_config_cluster.yml)  then rke up** 

- **panic: runtime error: invalid memory address or nil pointer dereference [recovered]**

INFO[0039] [certificates] Successfully deployed kubernetes certificates to Cluster nodes
INFO[0039] [reconcile] Reconciling cluster state
INFO[0039] [reconcile] Check etcd hosts to be deleted
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xd7998d]

goroutine 1 [running]:
github.com/rancher/rke/vendor/github.com/urfave/cli.HandleAction.func1(0xc0005bcca0)
        /go/src/github.com/rancher/rke/vendor/github.com/urfave/cli/app.go:472 +0x278
panic(0x135bca0, 0x2347070)
        /usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/rancher/rke/vendor/k8s.io/client-go/util/cert.EncodeCertPEM(0x0, 0xc0003871d0, 0x17338c0, 0x2383be0)
        /go/src/github.com/rancher/rke/vendor/k8s.io/client-go/util/cert/pem.go:69 +0x4d
github.com/rancher/rke/cluster.reconcileEtcd(0x1755fe0, 0xc00003e058, 0xc00052e900, 0xc00052e000, 0xc0002ea870, 0x0, 0x0)
        /go/src/github.com/rancher/rke/cluster/reconcile.go:175 +0x130
github.com/rancher/rke/cluster.ReconcileCluster(0x1755fe0, 0xc00003e058, 0xc00052e000, 0xc00052e900, 0x0, 0x0, 0x1526bfb, 0xb, 0x0, 0x0, ...)
        /go/src/github.com/rancher/rke/cluster/reconcile.go:43 +0x1ab
github.com/rancher/rke/cmd.ClusterUp(0x1755fe0, 0xc00003e058, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1526bfb, 0xb, 0x0, ...)
        /go/src/github.com/rancher/rke/cmd/up.go:133 +0x557
github.com/rancher/rke/cmd.clusterUpFromCli(0xc00025eb40, 0x0, 0x0)
        /go/src/github.com/rancher/rke/cmd/up.go:242 +0x582
reflect.Value.call(0x12f2040, 0x15c5678, 0x13, 0x151f271, 0x4, 0xc0004cec40, 0x1, 0x1, 0xc00007c080, 0x2, ...)
        /usr/local/go/src/reflect/value.go:447 +0x449
reflect.Value.Call(0x12f2040, 0x15c5678, 0x13, 0xc0004cec40, 0x1, 0x1, 0x5, 0x4, 0xc000418a08)
        /usr/local/go/src/reflect/value.go:308 +0xa4
github.com/rancher/rke/vendor/github.com/urfave/cli.HandleAction(0x12f2040, 0x15c5678, 0xc00025eb40, 0x0, 0x0)
        /go/src/github.com/rancher/rke/vendor/github.com/urfave/cli/app.go:481 +0x1fb
github.com/rancher/rke/vendor/github.com/urfave/cli.Command.Run(0x151ecc2, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x153078a, 0x14, 0x0, ...)
        /go/src/github.com/rancher/rke/vendor/github.com/urfave/cli/command.go:186 +0x8f6
github.com/rancher/rke/vendor/github.com/urfave/cli.(*App).Run(0xc00041a600, 0xc000038090, 0x3, 0x3, 0x0, 0x0)
        /go/src/github.com/rancher/rke/vendor/github.com/urfave/cli/app.go:235 +0x52e
main.mainErr(0x173c160, 0xc00000c018)
        /go/src/github.com/rancher/rke/main.go:56 +0xf9e
main.main()
        /go/src/github.com/rancher/rke/main.go:19 +0x3e

**Seems can't lost cluster.yml and cluster.rkestate**

**Comment by vjtm** [comment]
Hi there
Trying out rke v0.2.8 I got the same error with the exact behaviour and error trace.

**Comment by shubb30** [comment]
We ran into this issue today.  The scenario was that we have a cluster that was originally provisioned by a co-worker using RKE, so he had the original rkestate file.  Yesterday, I upgraded the Kubernetes version by running RKE against the cluster.  The upgrade went fine, but then today we noticed from the UI, that Rancher could not see the etcd health and it showed that etcd was unavailable.  All of the workloads in the cluster were still working, but Rancher could not check the etcd health.

We tried running `rke up` again, and we started seeing the same `remote error: tls: bad certificate` errors described here.  We tried renaming the `/etc/kubernetes/ssl` directory on each node, and then running rke again, with no luck.  We tried rebooting the nodes with no luck.

What we finally did that fixed it was to run `rke cert rotate` against the cluster, and that was able to correctly re-generate the certificates, and then the rke up finished and the cluster came back up healthy again.

YMMV, but I'd suggest backing up the certs directory first.
